go-backend/service/organization_http.go

```diff
@@ -0,0 +1,203 @@
+package service
+
+import (
+	"encoding/json"
+	"joshsoftware/peerly/db"
+	"net/http"
+	"strconv"
+	"github.com/gorilla/mux"
+	logger "github.com/sirupsen/logrus"
+)
+
+// @Title listOrganizationHandler
+// @Description list all Organizations
+// @Router /organizations [get]
+// @Accept  json
+// @Success 200 {object}
+// @Failure 400 {object}
+func listOrganizationHandler(deps Dependencies) http.HandlerFunc {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		organizations, err := deps.Store.ListOrganizations(req.Context())
+		if err != nil {
+			logger.WithField("err", err.Error()).Error("Error fetching data")
+			rw.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		respBytes, err := json.Marshal(organizations)
+		if err != nil {
+			logger.WithField("err", err.Error()).Error("Error marshaling organizations data")
+			rw.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		rw.Header().Add("Content-Type", "application/json")
+		rw.Write(respBytes)
+	})
+}
+
+// @Title createOrganizationHandler
+// @Description Create Organizations
+// @Router /organizations [post]
+// @Accept  json
+// @Success 200 {object}
+// @Failure 400 {object}
+func createOrganizationHandler(deps Dependencies) http.HandlerFunc {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		var organization db.Organization
+		err := json.NewDecoder(req.Body).Decode(&organization)
+		if err != nil {
+			rw.WriteHeader(http.StatusInternalServerError)
+			logger.WithField("err", err.Error()).Error("Error while decoding organizations data")
+			return
+		}
+		errorResponse, valid := organization.ValidateOrganization()
+		if valid == false {
+			respBytes, err := json.Marshal(errorResponse)
+			if err != nil {
+				logger.WithField("err", err.Error()).Error("Error marshaling organizations data")
+				rw.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+	
+			rw.Header().Add("Content-Type", "application/json")
+			rw.WriteHeader(http.StatusPreconditionFailed)
```

Comment:
must be bad request. 412 has a different use case i think. https://httpstatuses.com/412

Reply:
We had returned status as `http.StatusBadRequest` earlier, but our wiki page mentions 412 as validation error and so we changed it. Could you tell a scenario in which might return 412, just out of curiosity?